dfs/dfs.go

```diff
@@ -54,7 +54,7 @@ type DFS interface {
 	// Backup saves and exports the current state of the system
 	Backup(info BackupInfo, w io.Writer) error
 	// Restore restores the system to the state of the backup
-	Restore(r io.Reader) (*BackupInfo, error)
+	Restore(r io.Reader, backupInfo *BackupInfo) error
```

Comment:
Because we can extract the BackupInfo most cheaply from the file, rather than from the stream. Otherwise we have to recreate the stream after getting out the info. It seemed silly to pull it out to check the version and then pull it out again inside the restore.